protocol/dataunit/client.go

```diff
@@ -1,49 +1,61 @@
 package dataunit
 
 import (
+	"context"
+	"io"
 	"sync"
 )
 
 // Client provides an ordered queue of client requests on a data unit connection.
-// Each call to ExchangeDataUnit will block until the peer responds or the underlying connection is closed.
-// Requests will be processed in strict FIFO order.
+// Each call to ExchangeDataUnit will block until the peer responds, the Context is canceled,
+// or the underlying connection is closed. Requests will be processed in strict FIFO order.
```

Comment:
"Requests will be processed in strict FIFO order" seems like it belongs here, but the details of `ExchangeDataUnit` seem like they should live with that function.